### PR TITLE
Fix tab controls overflowing right content at max ui scale

### LINKED ISSUE #4415: hamburger menu not visible in chat if too many chats are open
osu_2019-03-05_21-10-45

### PR DESCRIPTION
**Pull request by Joehuu**
Also fixes https://github.com/ppy/osu/issues/4415.

### CODE REVIEW

**Comment by bdach** [comment]
For visual reference:

| Before | After |
| :-: | :-: |
| osu_2019-10-06_19-57-30 | osu_2019-10-06_19-58-41 |
| osu_2019-10-06_20-02-33 | osu_2019-10-06_20-03-16 |

File: osu.Game/Screens/Select/BeatmapDetailAreaTabControl.cs [hunk 1]
```diff
@@ -69,6 +74,13 @@ private void load(OsuColour colour, OsuConfigManager config)
             tabs.Current.TriggerChange();
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            tabsContainer.Padding = new MarginPadding { Right = modsCheckbox.Width };
```
Comment (bdach) [inline osu.Game/Screens/Select/BeatmapDetailAreaTabControl.cs]: Just to clarify, is this done in `Update` because the checkbox changes width dynamically?

**Review by bdach — COMMENTED** [review]
Visually looks good which is probably enough seeing that this is a layout reorder, but I just want to clarify one thing.

**Comment by Aergwyn** [comment]
Was the second part necessary or an accident from the changes? It looked fine to me.

**Comment by bdach** [comment]
> Was the second part necessary or an accident from the changes? It looked fine to me.

I took that screenshot above where the overflow menu is behind the details container from current `master`, unsure if it was always broken or a regression.

On another note though I did notice that the on extreme scalings the `Selected Mods` checkbox item on leaderboard tabs breaks badly, I don't know if there is currently an easy way of fixing that, and probably out of scope anyways?

File: osu.Game/Screens/Select/BeatmapDetailAreaTabControl.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Screens/Select/BeatmapDetailAreaTabControl.cs]: Checking again it does change width when the the text gets bold when checked, so this is probably right.

Initially, I just didn't want to specify some random fixed number as the padding.

**Comment by peppy** [comment]
In the above screenshots, the "..." button seems to be moving. It shouldn't move (should remain right-aligned in all cases).

**Comment by Joehuu** [comment]
Went an old approach (fixed numbers), since I think using `Update` is bad. Fixed the concerns.

**Review by peppy — APPROVED** [review]

**Pull request merged**